src/components/user-list.js

```diff
@@ -26,67 +28,61 @@ export default class UserList extends React.Component {
         };
     }
 
-    _handleClickAddUser () {
+    _handleClickAddUser = () => {
         this.setState({ wantAdd: !this.state.wantAdd });
-    }
+    };
 
-    _onChange = (userList) => {
-        this.setState({ users: userList });
+    _onChange = () => {
+        this.setState(this._getAppState());
     };
 
-    _handleClickFindUser () {
+    _handleClickFindUser = () => {
         const { wantFind } = this.state;
 
-        if (wantFind) {
-            this.setState({ users: UserStore.getUsers() });
-        }
         this.setState({ wantFind: !wantFind });
+    };
+
+    _usersFound = (usersFound) => {
+        this.setState({ users: usersFound });
+    };
+
+    _handleClickStopSearch = () => {
+        this.setState(this._getAppState());
     }
 
     componentDidMount () {
         UserStore.on(EVENT_TYPE.change, this._onChange);
+        UserStore.on(EVENT_TYPE.usersFound, this._usersFound);
     }
 
     componentWillUnmount () {
         UserStore.removeListener(EVENT_TYPE.change, this._onChange);
+        UserStore.removeListener(EVENT_TYPE.usersFound, this._usersFound);
     }
 
     render () {
         const { wantAdd, users, wantFind } = this.state;
-
         return (
             <div className="UserList">
-
-                <button className="ButtonFindUser" onClick={this._handleClickFindUser}>
+                <button className="ButtonFindUser"
+                        onClick={wantFind ? this._handleClickStopSearch : this._handleClickFindUser}>
                     {wantFind ? 'Stop searching' : 'Find User'}
                 </button>
                 {
                     wantFind &&
                     (
-                        <FindUser/>
-                    )
-                }
-                {
-                    !wantFind &&
-                    (
-                        <button className="ButtonAddUser" data-testid="ButtonAddUser"
-                                onClick={this._handleClickAddUser}>
-                            Add new User
-                        </button>
-                    )
-                }
-                {
-                    !wantFind && wantAdd &&
-                    (
-                        <User isNewUser={true} />
+                        <SearchForm/>
```

Comment:
(minor)
We usually use brackets if a component is multilined, e.g.
```js
{
    foo &&
    (
        <Bar baz="baz" baz="baz" baz="baz"
                 baz="baz" baz="baz"
                 baz="baz" baz="baz">
        <input/>
       </Bar>
    )
}
```
If a component only occupies one line we can do no use them, e.g.
```js
{
    foo &&
    <Bar/>
}
```